experimental/devices/hd44780/hd44780.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package hd44780 controls the Hitachi LCD display chipset HD-44780
+//
+// Datasheet
+//
+// https://www.sparkfun.com/datasheets/LCD/HD44780.pdf
+package hd44780
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/gpio"
+)
+
+// lineTwo offset for the second line in the LCD buffer.
+const lineTwo = 0x40
+
+// Dev is the 4-bit addressing device for HD-44780
+type Dev struct {
+	// data pins
+	dataPins []gpio.PinOut
+
+	// register select pin
+	rsPin gpio.PinOut
+
+	// enable pin
+	enablePin gpio.PinOut
+}
+
+// New creates and initializes the LCD device
+//	data - references to data pins
+//	rs - rs pin
+//	e - strobe pin
+func New(data []gpio.PinOut, rs, e gpio.PinOut) (*Dev, error) {
```

Comment:
Number of lines should be provided as an argument (1 or 2).

Reply:
I don't think that it makes much difference here, there's nothing related to the line calculation in the code, and the client knows what exactly she has for the hardware side. Especialy it is not possible to figure out whether the device supports 1 line or 2, so the extra validation makes no sense.

Reply:
The application author and the end user are often different persons. If validation is skipped, it'd be good to note this in the documentation.